openfl-tutorials/experimental/106_Scikit_Learn_Linear_Regression_Workflow.ipynb

```diff
@@ -0,0 +1,551 @@
+    "        print(f'Bias: {self.model.intercept_}')\n",
+    "            "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "af89e7e5-6cfc-46bc-acd2-7d5bfb373091",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Define input array with angles from 60deg to 400deg converted to radians\n",
+    "x = np.array([i*np.pi/180 for i in range(60,400,4)])\n",
+    "np.random.seed(10)  # Setting seed for reproducibility\n",
+    "y = np.sin(x) + np.random.normal(0,0.15,len(x))\n",
+    "# plt.plot(x,y,'.')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ffefca2b-d7f6-4111-8872-c017c182a2de",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Initialize the model\n",
+    "lr_model = SklearnLinearRegressionLasso(n_feat=1, alpha=0.1)\n",
+    "\n",
+    "# Fit the model\n",
+    "lr_model.fit(x[:,np.newaxis], y)\n",
+    "\n",
+    "#print the final parameters\n",
+    "lr_model.print_parameters()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "410f2d80-989a-43ab-958f-7b68fd8f2e90",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# We can also solve this 1D problem using Numpy\n",
+    "numpy_solution = np.polyfit(x,y,1)\n",
+    "predictor_np = np.poly1d(numpy_solution)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6cb323db-9f3a-42af-94da-4b170adef867",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Predict using the model\n",
+    "y_hat = lr_model.predict(x)\n",
+    "# Plot the results\n",
+    "y_np = predictor_np(x)\n",
+    "plt.plot(x,y,'.')\n",
+    "plt.plot(x,y_hat,'.')\n",
+    "plt.plot(x,y_np,'--')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ffd4d2d7-5537-496a-88c1-301da87d979c",
+   "metadata": {},
+   "source": [
+    "## Now we run the same training on federated learning workflow api"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "08527aab-4b0f-472b-af0c-27ed4ade85c1",
+   "metadata": {},
+   "source": [
+    "## Import required libraries for federated learning"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1b3c0039-e1f7-4047-b98b-a2d4bd42f015",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Import ncessary libraries\n",
+    "import numpy as np\n",
+    "from sklearn.linear_model import Lasso\n",
+    "from sklearn.metrics import mean_squared_error\n",
+    "from sklearn.datasets import make_regression\n",
+    "from openfl.experimental.interface import FLSpec\n",
+    "from openfl.experimental.placement import aggregator, collaborator\n",
+    "from openfl.experimental.runtime import FederatedRuntime\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "from tensorflow.keras.utils import to_categorical\n",
```

Comment:
remove to_categorical and mnist

Reply:
resolved in e1e711efeb4f84ed325d0ae3401328c80502479a